src/Routing/QueryRouteEnhancer.php

```diff
@@ -122,4 +141,32 @@ protected function enhanceSingle($defaults, $values) {
       '_controller' => $defaults['_graphql']['single'],
     ];
   }
-}
\ No newline at end of file
+
+  /**
+   * @param $query
+   * @param $id
+   * @param $version
+   * @return null
+   */
+  protected function getQuery($query, $id, $version) {
```

Comment:
if we have a $query, just return it. If we don't have a $query, load by $id and $version.

```
if (!empty($query)) {
  return $query;
}

$queryMap = $this->entityTypeManager->getStorage('graphql_query_map')->load($version);
return !empty($queryMap[$id]) ? $queryMap[$id] : NULL;
```